src/DataStreams/RemoteQueryExecutor.h

```diff
@@ -93,7 +93,7 @@ class RemoteQueryExecutor
 
     const String query;
     String query_id = "";
-    Context context;
+    const Context & context;
```

Comment:
Is it really needed?

Reply:
Not "really needed", but why it should copy the context?
Lifetime of the passed Context should not exceed the RemoteQueryExecutor, so looks ok.
@nikitamikhaylov Are there any problems with this?

Reply:
There were some problems in Arcadia. I simply returned back the copy of Context here #18084

Reply:
@nikitamikhaylov Ok, I see, thanks!
Any chance that arcadia build can be fixed instead?

Reply:
>UPD: I think I'll be able to remove this " fix " later, after synchronization starts working.

Ok, got it